controllers/toolchainstatus/toolchainstatus_controller.go

```diff
@@ -272,53 +280,99 @@ func (r *Reconciler) synchronizeWithCounter(reqLogger logr.Logger, toolchainStat
 // hostOperatorHandleStatus retrieves the Deployment for the host operator and adds its status to ToolchainStatus. It returns false
 // if the deployment is not determined to be ready
 func (r *Reconciler) hostOperatorHandleStatus(reqLogger logr.Logger, toolchainStatus *toolchainv1alpha1.ToolchainStatus) bool {
+	// ensure host operator status is set
+	if reflect.ValueOf(toolchainStatus.Status.HostOperator).IsZero() {
+		toolchainStatus.Status.HostOperator = &toolchainv1alpha1.HostOperatorStatus{}
+	}
+
 	operatorStatus := &toolchainv1alpha1.HostOperatorStatus{
 		Version:        version.Version,
 		Revision:       version.Commit,
 		BuildTimestamp: version.BuildTime,
+		RevisionCheck:  toolchainv1alpha1.RevisionCheck{},
 	}
 	// look up name of the host operator deployment
-	hostOperatorName, err := commonconfig.GetOperatorName()
-	if err != nil {
-		reqLogger.Error(err, status.ErrMsgCannotGetDeployment)
+	hostOperatorName, errDeploy := commonconfig.GetOperatorName()
+	if errDeploy != nil {
+		reqLogger.Error(errDeploy, status.ErrMsgCannotGetDeployment)
 		errCondition := status.NewComponentErrorCondition(toolchainv1alpha1.ToolchainStatusDeploymentNotFoundReason,
-			fmt.Sprintf("%s: %s", status.ErrMsgCannotGetDeployment, err.Error()))
+			fmt.Sprintf("%s: %s", status.ErrMsgCannotGetDeployment, errDeploy.Error()))
 		operatorStatus.Conditions = []toolchainv1alpha1.Condition{*errCondition}
 		toolchainStatus.Status.HostOperator = operatorStatus
 		return false
 	}
 	hostOperatorDeploymentName := fmt.Sprintf("%s-controller-manager", hostOperatorName)
 	operatorStatus.DeploymentName = hostOperatorDeploymentName
 
+	// save host operator deployment check status in this flag, it will be returned in case no errors are found by the following checks.
+	noError := true
```

Comment:
Registration service returns immediately if there is a failure in the health check because without that I don't have the `revision` SHA to compare with the one in GitHub. While it's different for host-operator, I can still do the check since I have all the input in any case. Hopefully this makes sense.

Reply:
And actually I just noticed that I'm using `version.Commit` here https://github.com/codeready-toolchain/host-operator/pull/819/files#diff-dbe76ee9dfabe95c78e8d2e7c3efdf6526bfd4719168b8529c518129cb83a6acR905

while I should use what's coming back for the health check call: `healthValues.Revision`!

Reply:
> just a small suggestion - could we make the var name simpler so it doesn't contain a logical operator inside of the name? eg. allOK := true or failed := false

And thanks for the suggestion, I thought for some time about how to name that variable, but could find a better name 😆 . `allOK` makes more sense indeed!

Reply:
I've changed the variable name to `allOK` and changed the version check to use `healthValues.Revision` instead of `version.Commit` for reg service. Please see: [afac3d4](https://github.com/codeready-toolchain/host-operator/pull/819/commits/afac3d4884ccae631340c95f990a4e7c0eec95bf)

Reply:
ok, I understand now - I didn't notice that before :see_no_evil: Thanks for the explanation :-)